soil_api/models/soil_property.py

```diff
@@ -0,0 +1,196 @@
+    ocs = "t/ha"
+    phh2o = "pH*10"
+    sand = "g/kg"
+    silt = "g/kg"
+    soc = "dg/kg"
+
+
+class SoilProperty(BaseModel):
+    value: int | str = Field(
+        ..., description="The value of the queried soil property", example=50
+    )
+    unit: str = Field(
+        ..., description="The unit of the queried soil property", example="g/kg"
+    )
+    property: SoilPropertiesNames = Field(
+        ..., description="The name queried soil property", example="Soil organic carbon"
+    )
+
+
+class SoilLayerList(BaseModel):
+    layers: List[SoilLayer] = Field(..., description="The queried soil property layers")
+
+
+class SoilLayer(BaseModel):
+    code: SoilPropertiesCodes = Field(
+        ..., description="The soil property code", example="bdod"
+    )
+    name: SoilPropertiesNames = Field(
+        ..., description="The name of the soil property", example="Bulk density"
+    )
+    unit: SoilPropertiesUnits = Field(
+        ..., description="The unit of the soil property", example="g/kg"
+    )
+    depths: List[SoilDepth] = Field(
+        ..., description="The queried soil depths with values"
+    )
+
+
+class SoilPropertyValues(BaseModel):
+    mean: float | None = Field(
+        None, description="The mean value of the soil property", example=50
+    )
+    Q0_05: float | None = Field(
+        None,
+        description="The 5th percentile of the soil property",
+        example=40,
+        alias="Q0.05",
+    )
+    Q0_5: float | None = Field(
+        None,
+        description="The 50th percentile of the soil property",
+        example=50,
+        alias="Q0.5",
+    )
+    Q0_95: float | None = Field(
+        None,
+        description="The 95th percentile of the soil property",
+        example=60,
+        alias="Q0.95",
+    )
+    uncertainty: float | None = Field(
+        None, description="The uncertainty of the soil property", example=5
+    )
+
+
+class SoilDepthLabels(Enum):
+    depth_0_5 = "0-5cm"
+    depth_0_30 = "0-30cm"
+    depth_5_15 = "5-15cm"
+    depth_15_30 = "15-30cm"
+    depth_30_60 = "30-60cm"
+    depth_60_100 = "60-100cm"
+    depth_100_200 = "100-200cm"
+
+
+# Create a reverse lookup dictionary
+REVERSE_DEPTH_LOOKUP = {
+    v.value: k for k, v in list(SoilDepthLabels.__members__.items())
+}
+
+
+# Function to get the enum member using the reverse lookup
+def get_soil_depth_from_label(depth_string: str) -> SoilDepthLabels:
+    try:
+        return REVERSE_DEPTH_LOOKUP[depth_string]
+    except KeyError:
+        raise ValueError(f"No soil depth label found for '{depth_string}'")
+
+
+class SoilDepthTop(Enum):
+    depth_0_5 = 0
+    depth_0_30 = 0
+    depth_5_15 = 5
+    depth_15_30 = 15
+    depth_30_60 = 30
+    depth_60_100 = 60
+    depth_100_200 = 100
+
+
+class SoilDepthBottom(Enum):
+    depth_0_5 = 5
+    depth_0_30 = 30
+    depth_5_15 = 15
+    depth_15_30 = 30
+    depth_30_60 = 60
+    depth_60_100 = 100
+    depth_100_200 = 200
+
+
+class SoilDepthUnits(Enum):
```

Comment:
In SoilGrids' own [API](https://rest.isric.org/soilgrids/v2.0/docs) they do this. We could of course drop it, but maybe SoilGrids envisions that there could be other units in the future? Here's an example response from their API:
```
{
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [
      9.51,
      60.1
    ]
  },
  "properties": {
    "layers": [
      {
        "name": "bdod",
        "unit_measure": {
          "d_factor": 100,
          "mapped_units": "cg/cm³",
          "target_units": "kg/dm³",
          "uncertainty_unit": ""
        },
        "depths": [
          {
            "range": {
              "top_depth": 0,
              "bottom_depth": 5,
              "unit_depth": "cm"
            },
            "label": "0-5cm",
            "values": {
              "Q0.05": 10
            }
          }
        ]
      }
    ]
  },
  "query_time_s": 0.18471789360046387
}
```

Reply:
You will need to add the new units to the enum as well if SoilGrids adds new units right? I don't understand how SoilDepthUnits-enum is more future-proof at the moment. What problem do you hope to solve using SoilDepthUnits?

Reply:
Ok. But even without the `SoilDepthsUnits` enum, if SoilGrids adds a new unit, we would have to modify the code to take this into account (the OpenEPI API is not making calls to the SoilGrids API), right? Using the enum, a `DepthRange` can be instantiated like this (it facilitates the mapping from the depth to the unit):

```
depth = "depth_5_15"
depth_range = DepthRange(
        top_depth=SoilDepthTop.__members__[depth],
        bottom_depth=SoilDepthBottom.__members__[depth],
        unit_depth=SoilDepthUnits.__members__[depth],
)
```
If a new depth were to be added, such as "0-5m", it would have to be added to the `SoilDepthLabels`, `SoilDepthTop`, `SoilDepthBottom`, and `SoilDepthUnits` enum, but the rest of the code could be left unchanged.